nethttp/server.go

```diff
@@ -43,6 +43,9 @@ func MWComponentName(componentName string) MWOption {
 	}
 }
 
+// SpanOption controls the behavior of the Middleware to the span.
+type SpanOption func(span opentracing.Span, r *http.Request)
```

Comment:
Is this ok? Just take request and span as input, do anything in the function option.

Or taking MWOption type of parameter to set tag to the span is better?

Reply:
1. SpanObserver might be a better name
2. create a new MWOption that accepts SpanObserver function, so that it can be passed to Middleware() instead of creating another MiddlewareWithSpanOption() function

Reply:
MWOption looks like only can be used to set operationName or ext.Component for span?
Sorry may not understand what you mean, Middleware() how to use SpanObserver as MWOption.

Reply:
You need to add a new function that creates a MWOption so that it can be passed to the main Middleware() function. It's known as a functional option pattern - https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis.